avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1402,14 +1509,14 @@ else if (schema.type() == Schema.Type.INT32)
   private static String unionMemberFieldName(org.apache.avro.Schema schema) {
     if (schema.getType() == org.apache.avro.Schema.Type.RECORD ||
         schema.getType() == org.apache.avro.Schema.Type.ENUM) {
-      return splitName(schema.getName())[1];
+      return schema.getFullName();
```

Comment:
Yeah, I get the issue but compatibility is a pain :( I left info about configs elsewhere. The goal would be to put this behind a config that we could eventually change to be on by default, but these types of changes require providing a window of time to deal with conversion/compatibility.